src/mlpack/methods/kmeans/elkan_kmeans_impl.hpp

```diff
@@ -144,18 +147,19 @@ double ElkanKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
           }
         }
       }
+      
+      // At this point, we know the new cluster assignment.
+      // Step 4: for each center c, let m(c) be the mean of the points
+      // assigned to c.
+      newCentroids.col(assignments[i]) += arma::vec(dataset.col(i));
+      counts[assignments[i]]++;
```

Comment:
Oh!  I see now.  This has moved into the `else` condition, so whether or not there is a `continue` in the `if` body is now irrelevant.  So, up to you whether you want to keep it like this and remove the `continue`, or revert this.  (Personally I think it is cleaner like this but without the `continue`, but, up to you which to do. :+1:)